static/skin/index.js

```diff
@@ -1,31 +1,65 @@
 const root = $(`link[type='root']`).attr('href');
+let viewPortHeight = window.innerHeight;
+let isFetching = false;
+let isEnd = false;
+const basicConfig = {
+    'start': 0,
+    'count': Math.floor(viewPortHeight/100 + 1) * 3
+};
+
+function queryUrlBuilder() {
+    let url = `${root}/catalog/search?`;
+    Object.keys(basicConfig).forEach((key, idx) => {
+        url += `${key}=${basicConfig[key]}${idx !== Object.keys(basicConfig).length - 1 ? '&' : ''}`;
+    });
+    return url;
```

Comment:
Using `map()` + `join()` is simpler:

```
url += Object.keys(basicConfig).map(key => `${key}=${basicConfig[key]}`).join("&")
```

Reply:
@veloman-yunkan we can convert it to a oneliner like this 
```
return `${root}/catalog/search?${Object.keys(basicConfig).map(key => `${key}=${basicConfig[key]}`).join("&")}`;
```
or will it be too complex?

Reply:
I personally stick to "the max line length of 80 characters" rule and would instead take measures to shorten the line proposed in my previous comment, instead of pursuing a longer and a more incomprehensible one-liner.